Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -1168,7 +1350,122 @@ namespace xr
                         inputSource.AimSpace.Pose.Orientation.W = location.pose.orientation.w;
                     }
                 }
-                
+
+                {
+                    // Initialize the gamepad object components
+                    auto& inputSource = InputSources[idx];
+                    auto controllerInfo = sessionImpl.ControllerInfo;
+                    if (inputSource.GamepadObject.buttons.size() != controllerInfo.CONTROLLER_BUTTONS)
+                    {
+                        inputSource.GamepadObject.buttons = std::vector<Button>(controllerInfo.CONTROLLER_BUTTONS);
+                    }
+                    if (inputSource.GamepadObject.axes.size() != controllerInfo.CONTROLLER_AXES)
+                    {
+                        inputSource.GamepadObject.axes = std::vector<float>(controllerInfo.CONTROLLER_AXES);
+                    }
+                    
+                }
+
+                // Get trigger value data
+                {
+                    // query input action state
+                    XrActionStateFloat triggerState{XR_TYPE_ACTION_STATE_FLOAT};
+                    XrActionStateGetInfo getInfo{XR_TYPE_ACTION_STATE_GET_INFO};
+                    getInfo.action = actionResources.ControllerGetTriggerValueAction;
+                    XrCheck(xrGetActionStateFloat(session, &getInfo, &triggerState));
+                    auto& inputSource = InputSources[idx];
+                    if (triggerState.changedSinceLastSync)
```

Comment:
nit: it seems like throughout the inputSource could be declared inside the changedSinceLastSync checks

Reply:
I'm curious on what difference that would make?

Reply:
I think its mostly to keep related code in one place. So since inputSource and controllerInfo aren't used outside of the conditional, moving their declaration inside the conditional would make it visually clearer where they are being used.